docs/adr/adr_authorization-specification.md

```diff
@@ -0,0 +1,173 @@
+
+#### Usergroups
+
+A usergroup reflects the user's responsibilities in the partner organisations. Currently, these usergroups exist:
+
+- administrator (head of operations)
+- coordinator
+- warehouse volunteer
+- free shop volunteer
+- library volunteer
+- label creator
+- God user
+
+#### Permissions
+
+Depending on the user group, a user is able to perform certain actions. In our authorization concept, these actions are mirrored by action-based permissions (ABP).
+ABPs represent functions in the application that the user who was granted the ABPs can execute. Examples: viewing the inventory, or managing products. The name of an ABP consists of a verb indicating the action, and a plural noun (if applicable), separated by underscore: `manage_products`, `view_inventory`, etc.
+
+In the application back-end however it needs to be distinguished which data a user is allowed to access in what way. This is achieved through resource-based permissions (RBP).
+An RBP refers to a resource in the database, and the methods that the user who was granted the RBP can execute. These methods correspond to database operations:
+
+- read: `SELECT`
+- create: `INSERT`
+- edit: `UPDATE`
+- write: `INSERT`, `UPDATE`
+- delete: `DELETE`
+- assign: `INSERT` into cross-reference table
+
+The naming convention for RBP is a singular noun (the resource; multi-word nouns concatenated by underscore), and a method name, separated by colon: `user:edit`, `beneficiary:create`, `tag_relation:read`, etc.
+
+Every ABP comprises one or more RBP, e.g. the ABP `manage_tags` stands for `tag:write`, `stock:read`, `tag_relation:read`, and `beneficiary:read`.
+
+The mapping of usergroup to ABPs, and ABP to RBPs is listed in [this document](https://docs.google.com/spreadsheets/d/1W4YWcc59wUFUWgReumdH6DQ4zU7JcTgvf6WEbdqaGHQ/edit#gid=0).
+
+### User management in Auth0
+
+[Auth0](https://auth0.com) is a service for managing user authentication and authorization. It serves as single source of truth. An authenticated user gets issued a JSON Web Token (JWT) holding their authorization information. For signing the JWT we use the RS256 algorithm: the token will be signed with our private signing key and can be verified using our public signing key.
+
+#### Reasons to use Auth0
+
+Cf. [related ADR](./docs/adr/adr_auth0.md)
+
+#### User entity in Auth0
+
+Any user registered for boxtribute has their authorization data (`app_metadata`) stored in the Auth0 database. The user attributes are:
+
+- `is_god`: `"1"` for god user, `"0"` otherwise
+- `usergroup_id`: the ID of the usergroup the user belongs to
+- `base_ids`: a list of base IDs that the user has access to
+- `organisation_id`: the ID of the organisation the user belongs to
+
+During registration, the user gets assigned a role, indicating their usergroup and the bases they belong to. The role is named like `base_1_coordinator`.
```

Comment:
@vahidbazzaz can you confirm that my understanding is correct? Does the assignment happen manually by an admin, or automatically?

Reply:
it happens manually. In dropapp the user is created and one user group can be assigned. Based on this selection the roles are added to this user in auth0. I will describe it in more detail in the dropapp section.